pybleau/app/ui/filter_expression_editor.py

```diff
@@ -1,31 +1,285 @@
+        self.source_df = source_df
+        if "included_cols" in traits:
+            # if any of the specified columns have value dtypes that are
+            # numerical, then throw an error
+            included = traits["included_cols"]
+            if not set(included).issubset(sorted(source_df)):
+                msg = f"The source_df does not contain all columns in the " \
+                      f"included_cols list."
+                logger.exception(msg)
+                raise KeyError(msg)
+            self.included_cols = traits["included_cols"]
+        self._create_trait_translators()
+        self._create_col_name_scrollable_view()
+        self._create_included_col_scroll_list()
+        self._edit_initialized = True
+
     def traits_view(self):
+        # make the view
         view = self.view_klass(
-            VGroup(
-                Item("exp", style="custom", show_label=False),
-                show_border=True
+            HSplit(
+                VGroup(
+                    Tabbed(
+                        Item("_scrollable_column_names", style="custom",
+                             editor=InstanceEditor(), show_label=False,
+                             label="Column Names"),
+                        *self._included_col_scroll_list
+                    ),
+                    HGroup(
+                        Item('click_type', style='custom', show_label=False),
+                        Spring(),
+                        label="When clicking a button...",
+                        show_border=True
+                    )
+                ),
+                VGroup(
+                    Item("expr", style="custom", show_label=False, width=450),
+                    show_border=True
+                )
             ),
             resizable=True,
             buttons=OKCancelButtons,
             title="Edit the filter...",
+            statusbar=StatusItem(name="status_bar_message"),
         )
         return view
 
+    def _create_trait_translators(self):
+        """ Create translators that associate column names/values with traits
+        """
+        # create for each column name
+        self._add_values_to_translators(sorted(self.source_df))
+        # create for each value in the included columns
+        for column in self.included_cols:
+            column_values = list(self.source_df[column].unique())
+            self.included_col_values[column] = column_values
+            self._add_values_to_translators(column_values)
+
+    def _add_values_to_translators(self, column_values: Iterable):
+        """ Add values from `column_values` to two translation dicts
+        """
+        for name in column_values:
+            # make a traited-version of the value
+            trait_name = f"trait_{sanitize_string(name)}"
+            # track the traited name with the original name key
+            self.traited_names[name] = trait_name
+            # track the original name with a traited name key
```

Comment:
[General advice] Comments and documentation is super important. But a line of comment for every line of code is a bit too much documentation. Ideally, the code itself is written so it's readable and your comments don't have to be quite that numerous. If they are, it lengthens the source files and makes it not ideal to build a global picture of the code.